hack/images/dcos/Makefile

```diff
@@ -1,65 +1,219 @@
-mkfile_path     := $(abspath $(lastword $(MAKEFILE_LIST)))
-current_dir     := $(patsubst %/,%,$(dir $(mkfile_path)))
-
-FROM ?= $(current_dir)/_build/k8sm
-TARGET_OBJ = km kubectl
-SOURCE_OBJ = $(TARGET_OBJ:%=$(FROM)/%)
-
+# input variables
+GIT_REF ?= v1.0.2-v0.6.1
+GIT_URL ?= https://github.com/mesosphere/kubernetes.git
+DOCKER_ORG ?= mesosphere
+KUBE_ROOT ?=
+SUDO ?=
+POSTFIX ?= -alpha
+
+# fixed values
+SHELL := /bin/bash
+
+GIT_REF_MINOR = $(lastword $(subst -, ,$(GIT_REF)))
+GIT_REF_MAJOR = $(firstword $(subst -, ,$(GIT_REF)))
+
+BUILD_DIR = _build
+
+ifneq ($(wildcard $(BUILD_DIR)),)
+	UNCLEAN_INFIX := -unclean
```

Comment:
You can even have something dirty **and** unclean. We have full support for that in the Makefile.